src/features/domain-settings/hooks/useDomainSettingsForm.ts

```diff
@@ -0,0 +1,223 @@
+	useWeb3,
+	useZnsSdk,
+} from '../../../lib/hooks';
+import { useTransaction } from '@zero-tech/zapp-utils/hooks/useTransaction';
+
+import { Step } from '@zero-tech/zui/components';
+
+export type UseDomainSettingsFormFormReturn = {
+	stepId: string;
+	formHeader: string;
+	errorText: string;
+	confirmActionType: ConfirmActionType;
+	loadingStatusText: string;
+	isTransactionLoading: boolean;
+	onStepUpdate: (step: Step) => void;
+	onTitleUpdate: (title: string) => void;
+	onFormDetailsSubmit: (values: FieldValues) => void;
+	onLockMetadataStatus: () => void;
+	onSetAndLockMetadata: () => void;
+	onSetMetadata: () => void;
+	onConfirmActionUpdate: (action: ConfirmActionType) => void;
+};
+
+export const useDomainSettingsForm = (
+	zna: string,
+): UseDomainSettingsFormFormReturn => {
+	const sdk = useZnsSdk();
+
+	const { account, provider } = useWeb3();
+	const { executeTransaction } = useTransaction();
+
+	// add isLoading to form
+	const { domainId, metadata, metadataLockedStatus } =
+		useDomainSettingsData(zna);
+
+	const [stepId, setStepId] = useState(steps[0].id);
+	const [errorText, setErrorText] = useState<string>();
+	const [loadingStatusText, setLoadingStatusText] = useState<string>();
+	const [formHeader, setFormHeader] = useState<string>('My Domain Settings');
+	const [isTransactionLoading, setIsTransactionLoading] = useState<boolean>();
+	const [confirmActionType, setConfirmActionType] =
+		useState<ConfirmActionType>();
+
+	const [details, setDetails] = useState<DetailsFormSubmit>();
+	console.log(details);
+	console.log(metadata);
+
+	// Set metadata form details
+	const onFormDetailsSubmit = ({
+		title,
+		description,
+		isMintable,
+		isBiddable,
+		gridViewByDefault,
+		customDomainHeader,
+		customDomainHeaderValue,
+	}: FieldValues): void => {
+		setDetails({
+			attributes: metadata?.attributes,
+			title,
+			description,
+			image: metadata?.image,
+			image_full: metadata?.image_full,
+			previewImage: metadata?.previewImage,
+			animation_url: metadata?.animation_url,
+			stakingRequests: metadata?.stakingRequests,
+			isBiddable,
+			isMintable,
+			gridViewByDefault,
+			customDomainHeader,
+			customDomainHeaderValue,
+		});
+	};
+
+	const onStepUpdate = (step: Step) => {
+		setStepId(step.id);
+	};
+
+	const onConfirmActionUpdate = (action: ConfirmActionType): void => {
+		setConfirmActionType(action);
+	};
+
+	const onTitleUpdate = (title: string): void => {
+		setFormHeader(title);
+	};
+
+	const onCheckMetadataLockStatus = useCallback(async () => {
+		const isDomainMetadataLocked = await sdk.isDomainMetadataLocked(
+			domainId,
+			provider.getSigner(),
+		);
+
+		return isDomainMetadataLocked;
+	}, [domainId, provider]);
+
+	const onGetDomainMetadata = useCallback(async () => {
+		const domainMetadata = await sdk.getDomainMetadata(
+			domainId,
+			provider.getSigner(),
+		);
+
+		console.log('here');
+
+		console.log('DOMINA', domainMetadata);
+
+		return domainMetadata;
+	}, [domainId, provider]);
```

Comment:
I wonder if instead of getting re-scraping lock status/metadata, we could invalidate the existing metadata query for the domain? That way it would populate the new domain lock status/metadata everywhere, rather than just the advanced metadata modal.

Reply:
Yes that would be ideal! Do you have a suggestion of the best way to invalidate the existing metadata. I'm assuming this will then update what the latest data should be ?

Reply:
As that's one of the issues currently when reaching the final step of the form, or closing the form once finished

Reply:
Might be worth pulling this branch and giving it a run. Seems to go well locally for me, would be good to verify that though!

Reply:
I do! I did a [similar thing in the Staking zApp](https://github.com/zer0-os/zApp-Staking/blob/master/src/features/stake/useStake.ts), because I needed to refresh staking data after a successful stake/unstake. In short, you just need to invalidate the `react-query` keys you want to reset, and you do it directly on the query client.

Something like this:
```js
const queryClient = useQueryClient();

const onUpdateMetadata = () => {
    queryClient.invalidateQueries(['key ']);
}
```

There is probably better ways to do this, as it looks like `useQuery` also returns `refetch` and `remove` functions. Maybe have a dig into those, and if they don't yield any results, use what I chucked above. The only flaw with above is it's slightly verbose, and the invalidation key has to be identical to the query key.